### DISCUSSION TITLE
Get records of specific message type in polymorphic stream

### DISCUSSION THREAD

**Discussion by ShivaniSK** [finos/TimeBase-CE#262] [Q&A] [answered]
Hello @alex-karpovich ,
We have below polymorphic stream created 
```

DURABLE STREAM "test_poly_stream" (
    CLASS "DBTrade" 'test db trade' (
        "price" '' FLOAT NOT NULL BINARY (32),
        "venue" '' VARCHAR MULTILINE,
        "book" '' VARCHAR MULTILINE,
        "counterparty" '' VARCHAR MULTILINE
    );
    CLASS "MarketTrade" 'test market trades stream' (
        "price" '' FLOAT BINARY (32),
        "venue" '' VARCHAR MULTILINE
    );
)
OPTIONS (POLYMORPHIC; PERIODICITY = 'IRREGULAR'; HIGHAVAILABILITY = FALSE; UNIQUE = FALSE; STORAGEVERSION = '5')
```

Have inserted data for both the message types.
![image](https://github.com/finos/TimeBase-CE/assets/8392899/a0077a67-2ff8-4240-90ee-87a186c5e6a1)

Attached screen snip for the same.

When i query as below in TimebaseWebAdmin
```
SELECT DBTrade:price 
FROM "test_poly_stream" 
```

![image](https://github.com/finos/TimeBase-CE/assets/8392899/d00bf9d7-2d84-4469-b5c6-0f774f80b2d4)

I can see records with timestamp of MarketTrade message type as well with null data in the price column.
Is there a way to get only those records which we inserted for DBTrade message type?

Regards,
Shivani

**Comment by alex-karpovich** [comment, marked as answer]
@ShivaniSK yes, it's possible. you need to specify message type  in query 

https://kb.timebase.info/docs/development/qql/QQL%205.5/qql-tut-filter#filtering-by-object-type

```
SELECT DBTrade:price 
FROM "test_poly_stream" WHERE THIS is DBTrade
```
